### ISSUE TITLE
Add default icons for explorer panels with option to customize icon for enhanced UX

### ISSUE THREAD

**Issue by peterblazejewicz** [Microsoft/vscode#43990]
That's feature request.

TL;DR

The icon in explorer panel, especially where there is a lot of explorer panels, can help to navigate and locate important stuff. Add default icon and allow to customize it.

Can we have an option to customize explorer section headers? Or at least let the workspace/editor explorer comes with some visual hint.
Now when extensions can add sections to explorer, there is a visual clutter there, the most importatn (file explorer) not always comes expanded and focused.

Here is my current setup. I just cannot see where is the workspace (that's vscode-editorconfig btw):

![image](https://user-images.githubusercontent.com/14539/36397531-814fa8d2-15c3-11e8-868a-18d188cfb454.png)

Compare that to Rider:

![image](https://user-images.githubusercontent.com/14539/36397598-b6557016-15c3-11e8-8478-a62f916d31dd.png)

or Xcode:

![image](https://user-images.githubusercontent.com/14539/36397618-c8e05638-15c3-11e8-99bb-71cfd66a8939.png)

I'm always able to see what the explorer section is showing in Rider/Xcode. VSCode could ship with default icons for explorer and allow users to use extension icon in extension explorer section.

For example: I know that my blue book icon is for my opened workspace - constantly:

![image](https://user-images.githubusercontent.com/14539/36397868-a646bb20-15c4-11e8-9300-807bad1792df.png)

It could be the other icon, that's not the point here.

Thanks for the editor!

**Comment by equinusocio** [comment]
I think having a way to customise all workbench icons is important for a better UX.
